db/migrations/9.1.02.sql

```diff
@@ -0,0 +1,251 @@
+
+alter table stos
+add constraint "stos_creator_id_fkey" FOREIGN KEY (creator_id) REFERENCES identities(id) ON UPDATE CASCADE,
+add constraint "stos_offering_portfolio_id_fkey" FOREIGN KEY (offering_portfolio_id) REFERENCES portfolios(id) ON UPDATE CASCADE,
+add constraint "stos_raising_portfolio_id_fkey" FOREIGN KEY (raising_portfolio_id) REFERENCES portfolios(id) ON UPDATE CASCADE,
+add constraint "stos_venue_id_fkey" FOREIGN KEY (venue_id) REFERENCES venues(id) ON UPDATE CASCADE;
+
+/**
+ * STO data migration - 
+ * 
+ *   1. Populate all the STO being created with `FundraiserCreated` event
+ *   2. Find the last (if any) `FundraiserWindowModified` event for all the STOs and update the corresponding `start` and `end` of the STO
+ *   3. Find the last modified status (if any) based on the events `FundraiserFrozen`, `FundraiserUnfrozen`, `FundraiserClosed`. This provides the latest status of all the STOs
+ *   4. To make the update data consistent, find the last event which triggered the STO modification and update the `block_id` and `updated_at` from the event against the corresponding STO
+ */
+
+-- Create all STOs
+with sto_data as (
+  select 
+    event_arg_1::int as sto_id,
+    event_arg_2 as name,
+    coalesce(
+        attributes->3->'value'->>'venueId',
+        attributes->3->'value'->>'venue_id' --needed for chain < 5.0.0
+      ) as venue_id,
+    coalesce(
+        attributes->3->'value'->>'offeringAsset',
+        attributes->3->'value'->>'offering_asset' --needed for chain < 5.0.0
+      ) as offering_asset_id,
+    coalesce(
+        attributes->3->'value'->'offeringPortfolio',
+        attributes->3->'value'->'offering_portfolio' --needed for chain < 5.0.0
+      ) as offering_portfolio,
+    coalesce(
+        attributes->3->'value'->>'raisingAsset',
+        attributes->3->'value'->>'raisiing_asset' --needed for chain < 5.0.0
+      ) as raising_asset_id,
+    coalesce(
+        attributes->3->'value'->'raisingPortfolio',
+        attributes->3->'value'->'raising_portfolio' --needed for chain < 5.0.0
+      ) as raising_portfolio,
+    "attributes"->3->'value'->>'creator' as creator_id,
+    to_timestamp(("attributes"->3->'value'->>'start')::NUMERIC/1000) as start,
+    to_timestamp(("attributes"->3->'value'->>'end')::NUMERIC/1000) as end,
+    "attributes"->3->'value'->'tiers' as tiers,
+    "attributes"->3->'value'->>'status' as status,
+    coalesce(
+        attributes->3->'value'->'minimumInvestment',
+        attributes->3->'value'->'minimum_investment' --needed for chain < 5.0.0
+      )::NUMERIC as minimum_investment,
+    block_id as created_block_id, 
+    block_id as updated_block_id, 
+    created_at as created_at, 
+    updated_at as updated_at
+  from events e
+  where module_id = 'sto' and event_id in ('FundraiserCreated')
+  order by block_id::int
+)
+INSERT INTO "public"."stos" ("id", "sto_id", "name", "venue_id", "offering_asset_id", "offering_portfolio_id", 
```

Comment:
```suggestion
insert into "public"."stos" ("id", "sto_id", "name", "venue_id", "offering_asset_id", "offering_portfolio_id", 
```

seems like it would be more consistent with the smaller case. I think with syntax highlighting all upper case is a bit redundant and more annoying to type. OK to leave.

I think the `::NUMERIC` all upper is good since the case should stand out a bit